### PR TITLE
Authenticate and Authorize Users on Frontend

### PR DESCRIPTION
**Pull request by Pochetes**
# Authentication Frontend
- centralized all API requests to one file instead of calling them on each component,  minimizes redundancy and unnecessary re-renders
- designed `Login` and `Register` components to begin taking input
- built my own custom hook for each form that calls the respective API request and validates (somewhat, will work on next milestone) form input
- persisted logged in user data as well as their token until they're logged out. implemented using `localStorage` and fetched the user given a token each time a user logs in, logs out, and registers
- created global state of user data using `useContext`, realized `user` and `setUser` were being prop-drilled everywhere
https://www.loom.com/share/9e0d03e5d14b49f1b4520b4f12d05570

## Future Goals
- begin working on the rest of the UI (e.g. rendering listings, creating a pool, reservations, etc.)
- improve validation on register and login forms
- implement reservations backend (e.g. create a new reservation, fetch all reservations from listing, etc.)

### CODE REVIEW

File: swimster-ui/src/hooks/useLoginForm.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export const useLoginForm = () => {
+    const { user, setUser } = useAuthContext()
+    const navigate = useNavigate()
+    const { register, handleSubmit } = useForm()
+    const [isProcessing, setIsProcessing] = useState(false)
+    const [error, setError] = useState({})
+    const [form, setForm] = useState({})
+
+    useEffect(() => {
+        // if user is logged in, redirect them to Home
```
Comment (roeshanigos) [inline swimster-ui/src/hooks/useLoginForm.js:16]: let's also add comments across the whole codebase for consistency

File: swimster-ui/src/components/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                    />
+                    <input
+                        className={
+                            `${style.inputElement} ${style.fullInput}`
+                        }
+                        type="password"
+                        placeholder="Password"
+                        {...register("password")}
+                    />
+                    <br></br>
```
Comment (roeshanigos) [inline swimster-ui/src/components/Login/Login.jsx:31]: nit: next time please use a margin instead of hardcoding a white space! it's cleaner that way

**Review by roeshanigos — APPROVED** [review]
Overall minor comments nothing big. One thing to note is you should probably

File: swimster-ui/src/components/Login/Login.jsx (hunk 2 above)
Comment (Pochetes) [reply swimster-ui/src/components/Login/Login.jsx:31]: oh yeah I should do that

File: swimster-ui/src/hooks/useLoginForm.js (hunk 1 above)
Comment (Pochetes) [reply swimster-ui/src/hooks/useLoginForm.js:16]: got it

File: swimster-ui/src/components/App/App.jsx [hunk 3]
```diff
@@ -1,12 +1,54 @@
...
+    }
+
+    const token = localStorage.getItem("swimster_session_token")
+    if (token) {
+      apiClient.setToken(token)
+      fetchUser()
+    }
+  }, [])
+
+
```
[the remark this answers is not in the document]
Comment (Pochetes) [reply swimster-ui/src/components/App/App.jsx:34]: ✅

**Pull request merged**